### PR TITLE
cleanup: Remove ambigious _t postfixes in struct names

### PR DESCRIPTION
**Pull request by DipSwitch**
Fixes: #1102

Ready for CI set to give Murdock some work. Local test seemd fine.

### CODE REVIEW

File: sys/include/cbor.h [hunk 1]
```diff
@@ -138,7 +138,7 @@ extern "C" {
  * @sa cbor_clear
  * @sa cbor_destroy
  */
-typedef struct cbor_stream_t {
+typedef struct cbor_stream {
```
Comment (cgundogan) [inline sys/include/cbor.h]: is `cbor_stream` needed at all?

File: core/include/mutex.h [hunk 2]
```diff
@@ -34,7 +34,7 @@
 /**
  * @brief Mutex structure. Must never be modified by the user.
  */
-typedef struct {
+typedef struct mutex {
```
Comment (cgundogan) [inline core/include/mutex.h]: why?

File: sys/include/net/fib.h [hunk 3]
```diff
@@ -37,7 +37,7 @@ extern "C" {
 /**
  * @brief Routing Protocol (RP) message content to request/reply notification
  */
-typedef struct rp_address_msg_t {
+typedef struct rp_address_msg {
```
Comment (cgundogan) [inline sys/include/net/fib.h]: can be removed also?

File: sys/include/net/fib/table.h [hunk 4]
```diff
@@ -38,52 +38,52 @@ extern "C" {
...
     /** Pointer to the next shared generic address on the source route */
-    struct fib_sr_entry_t *next;
+    struct fib_sr_entry *next;
 } fib_sr_entry_t;
 
 /**
 * @brief Container descriptor for a FIB source route
 */
-typedef struct fib_sr_t {
+typedef struct fib_sr {
```
Comment (cgundogan) [inline sys/include/net/fib/table.h]: can be removed?

File: sys/include/net/fib/table.h [hunk 5]
```diff
@@ -38,52 +38,52 @@ extern "C" {
...
-    struct fib_sr_entry_t *sr_dest;
+    fib_sr_entry_t *sr_dest;
 } fib_sr_t;
 
 /**
 * @brief Container for one FIB source route table,
 *        combining source routes and an entry pool
 */
-typedef struct fib_sr_meta_t {
+typedef struct fib_sr_meta {
```
Comment (cgundogan) [inline sys/include/net/fib/table.h]: can be removed?

File: sys/include/universal_address.h [hunk 6]
```diff
@@ -56,7 +56,7 @@ extern "C" {
 /**
  * @brief The container descriptor used to identify a universal address entry
  */
-typedef struct universal_address_container_t {
+typedef struct universal_address_container {
```
Comment (cgundogan) [inline sys/include/universal_address.h]: can be removed?

File: sys/net/routing/nhdp/iib_table.h [hunk 7]
```diff
@@ -33,7 +33,7 @@ extern "C" {
 /**
  * @brief   Possible L_STATUS values of a link tuple
  */
-typedef enum iib_link_tuple_status_t {
+typedef enum iib_link_tuple_status {
```
Comment (cgundogan) [inline sys/net/routing/nhdp/iib_table.h]: can be removed? etc. etc.

File: core/include/mutex.h (hunk 2 above)
Comment (DipSwitch) [inline core/include/mutex.h]: All structs in core have a name and typedef.

Comment (cgundogan) [inline core/include/mutex.h]: for what reason? the typedef is sufficient if the struct name is nowhere used

**Comment by DipSwitch** [comment]
Addressed

**Comment by DipSwitch** [comment]
Fixed pthread compile errors, fixed doxygen warnings (which Travis reported and Murdock not, although Travis thought the warnings to be ok)

**Comment by haukepetersen** [comment]
Changes look valid to me, though Murdock is not really happy, yet. ACK once that is fixed.

File: sys/include/net/fib/table.h [hunk 8]
```diff
@@ -93,12 +93,12 @@ typedef struct fib_sr_meta_t {
 } fib_sr_meta_t;
 
 /**
-* @breif FIB table type for single hop entries
+* @brief FIB table type for single hop entries
```
Comment (cgundogan) [inline sys/include/net/fib/table.h]: nice catch (:

File: sys/net/routing/nhdp/nhdp.h [hunk 9]
```diff
@@ -88,7 +88,7 @@ extern "C" {
 /**
  * @brief   MANET interface representation
  */
-typedef struct nhdp_if_entry_t {
+typedef struct nhdp_if_entry {
```
Comment (cgundogan) [inline sys/net/routing/nhdp/nhdp.h]: is this name used somewhere?

File: sys/posix/pthread/include/pthread_cond.h [hunk 10]
```diff
@@ -36,7 +36,7 @@ extern "C" {
 /**
  * @note condition attributes are currently NOT USED in RIOT condition variables
  */
-typedef struct pthread_condattr_t {
+typedef struct pthread_condattr {
```
Comment (cgundogan) [inline sys/posix/pthread/include/pthread_cond.h]: can this be removed?

File: sys/posix/pthread/include/pthread_cond.h [hunk 11]
```diff
@@ -46,7 +46,7 @@ typedef struct pthread_condattr_t {
  *
  * @warning fields are managed by cv functions, don't touch
  */
-typedef struct pthread_cond_t {
+typedef struct pthread_cond {
```
Comment (cgundogan) [inline sys/posix/pthread/include/pthread_cond.h]: can this be removed?

File: sys/posix/pthread/include/pthread_cond.h (hunk 10 above)
Comment (DipSwitch) [inline sys/posix/pthread/include/pthread_cond.h]: yes

File: sys/posix/pthread/include/pthread_cond.h (hunk 11 above)
Comment (DipSwitch) [inline sys/posix/pthread/include/pthread_cond.h]: yes

File: sys/net/routing/nhdp/nhdp.h (hunk 9 above)
Comment (DipSwitch) [inline sys/net/routing/nhdp/nhdp.h]: nope

**Comment by DipSwitch** [comment]
Let's give Murdock another change =)

**Comment by DipSwitch** [comment]
And it agree's.

**Comment by kaspar030** [comment]
&go.

**Pull request merged**